include/mapper.h

```diff
@@ -19,19 +19,35 @@
 #ifndef DOSBOX_MAPPER_H
 #define DOSBOX_MAPPER_H
 
+#include "dosbox.h"
+
 #include <string>
 #include <vector>
-#include "setup.h"
-#include "types.h"
 
-enum MapKeys {
-	MK_f1,MK_f2,MK_f3,MK_f4,MK_f5,MK_f6,MK_f7,MK_f8,MK_f9,MK_f10,MK_f11,MK_f12,
-	MK_return,MK_kpminus,MK_scrolllock,MK_printscreen,MK_pause,MK_home
+#include <SDL.h>
 
-};
+constexpr uint32_t MMOD1 = 0x1;
+constexpr uint32_t MMOD2 = 0x2;
+constexpr uint32_t MMOD3 = 0x4;
 
 typedef void (MAPPER_Handler)(bool pressed);
-void MAPPER_AddHandler(MAPPER_Handler * handler,MapKeys key,Bitu mods,char const * const eventname,char const * const buttonname);
+
+/* Associate function handler with a key combination
+ *
+ * handler     - function to be triggered
+ * key         - SDL scancode for key triggering the event.
+ *               Use SDL_SCANCODE_UNKNOWN to skip adding default key binding.
+ * mods        - key modifier events used for this action (according to user
+ *               preferences); a bitmask of MMOD1, MMOD2, MMOD3 values.
+ * event_name  - event name used for serialization in mapper file
+ * button_name - human-visible event name used in mapper GUI
```

Comment:
Minor suggestion: _"button_name - descriptive event name visible in the mapper GUI"_